pageserver/src/tenant.rs

```diff
@@ -570,12 +705,66 @@ impl Tenant {
     /// finishes. You can use wait_until_active() to wait for the task to
     /// complete.
     ///
-    pub fn spawn_attach(
+    pub fn spawn_start_attach(
         conf: &'static PageServerConf,
         tenant_id: TenantId,
         remote_storage: GenericRemoteStorage,
         ctx: &RequestContext,
-    ) -> Arc<Tenant> {
+    ) -> Result<Arc<Tenant>, AttachError> {
+        // Commit attach operation to disk, then return.
+        // The actual work is done in the background.
+        // If we crash before it finishes, we'll resume on restart.
+        let tenant_dir = conf.tenant_path(&tenant_id);
+        let marker_file = conf.tenant_attaching_mark_file_path(&tenant_id);
+        if tenant_dir.exists() {
+            return Err(AttachError::LocalStateExists {
+                what: "tenant directory",
+                path: tenant_dir,
+            });
+        }
+        if marker_file.exists() {
+            return Err(AttachError::LocalStateExists {
+                what: "attaching marker file",
+                path: marker_file,
+            });
+        }
+
+        // create the marker file, i.e., /tenants/${tenant_id}___attaching
+        fs::File::create(&marker_file).map_err(|error| AttachError::CreateMarkerFile {
```

Comment:
```suggestion
        std::fs::File::create(&marker_file).map_err(|error| AttachError::CreateMarkerFile {
```

Probably better to move these as either `std` or `tokio` since we now have so much of both on the same path even.